pandas/core/reshape/concat.py

```diff
@@ -26,8 +28,27 @@
 # Concatenate DataFrame objects
 
 
+@overload
+def concat(
+    objs: Union[Sequence["DataFrame"], Mapping[Optional[Hashable], "DataFrame"]],
+    axis=0,
+    join: str = "outer",
+    ignore_index: bool = False,
+    keys=None,
+    levels=None,
+    names=None,
+    verify_integrity: bool = False,
+    sort: bool = False,
+    copy: bool = True,
+) -> "DataFrame":
+    ...
+
+
+@overload
 def concat(
-    objs,
+    objs: Union[
```

Comment:
should we not have overload this taking account axis=1 ? (means more overloads but each is much more strict)

Reply:
We can't untiloverload based on values until we can use Literals (py3.8). Until then we can only overload based on types.